staging/src/k8s.io/apimachinery/pkg/runtime/extension.go

```diff
@@ -18,10 +18,81 @@ package runtime
 
 import (
 	"bytes"
-	"encoding/json"
+	gojson "encoding/json"
 	"errors"
+	"fmt"
+	"reflect"
+
+	cbordirect "k8s.io/apimachinery/pkg/runtime/serializer/cbor/direct"
+	"k8s.io/apimachinery/pkg/util/json"
+	"sigs.k8s.io/structured-merge-diff/v4/value"
+
+	cbor "github.com/fxamacker/cbor/v2"
 )
 
+// RawExtension intentionally avoids implementing value.UnstructuredConverter for now because the
+// signature of ToUnstructured does not allow returning an error value in cases where the conversion
+// is not possible (content type is unrecognized or bytes don't match content type).
+func (re RawExtension) toUnstructured() (interface{}, error) {
+	if re.Raw != nil {
+		contentType, err := re.rawContentType()
+		if err != nil {
+			return nil, err
+		}
+
+		switch contentType {
+		default:
+			return nil, fmt.Errorf("cannot convert RawExtension with unrecognized content type %q to unstructured", contentType)
+		case ContentTypeJSON:
+			var u interface{}
+			if err := json.Unmarshal(re.Raw, &u); err != nil {
+				return nil, fmt.Errorf("failed to parse RawExtension bytes as JSON: %w", err)
+			}
+			return u, nil
+		case ContentTypeCBOR:
+			var u interface{}
+			if err := cbordirect.Unmarshal(re.Raw, &u); err != nil {
+				return nil, fmt.Errorf("failed to parse RawExtension bytes as CBOR: %w", err)
+			}
+			return u, nil
+		}
+	}
+
+	if re.Object != nil {
```

Comment:
It's interesting to me that we're favoring `Raw` over `Object` when both are set.  What's the rationale here?  If we don't expect them ever to both be set, should that case be an error?

Reply:
I can't imagine a reason why they would both be set intentionally. The preference for non-nil Raw is based entirely on the existing behavior of MarshalJSON, which ignores a non-nil Object if Raw is non-nil. I'm hesitant to introduce a new error condition because I don't know if there are existing programs that would break. Should I track this somewhere?

Reply:
SGTM. A comment that MarshalJSON set precedence here seems sufficient to me.